rbd/admin/admin_test.go

```diff
@@ -30,3 +31,94 @@ func TestRawLevelSpec(t *testing.T) {
 	rls = NewRawLevelSpec("totally! invalid! haha. ha...")
 	assert.Equal(t, "totally! invalid! haha. ha...", rls.spec)
 }
+
+func TestNewImageSpec(t *testing.T) {
+	type args struct {
+		pool      string
+		namespace string
+		image     string
+	}
+	tests := []struct {
+		name string
+		args args
+		want ImageSpec
+	}{
+		{
+			name: "onlyImageName",
+			args: args{
+				pool:      "",
+				namespace: "",
+				image:     "img",
+			},
+			want: ImageSpec{
+				spec: "img",
+			},
+		},
+		{
+			name: "Image&PoolName",
+			args: args{
+				pool:      "pool",
+				namespace: "",
+				image:     "img",
+			},
+			want: ImageSpec{
+				spec: "pool/img",
+			},
+		},
+		{
+			name: "all args",
+			args: args{
+				pool:      "pool",
+				namespace: "ns",
+				image:     "img",
+			},
+			want: ImageSpec{
+				spec: "pool/ns/img",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewImageSpec(tt.args.pool, tt.args.namespace, tt.args.image); !reflect.DeepEqual(got, tt.want) {
```

Comment:
Please use the testify helper functions for this. Testify gives standard/cookie cutter output for this and avoids hand rolling this kind of thing.